3scale/apisonator
# THREESCALE-9682: Add support for OpenTelemetry

This adds OpenTelemetry instrumentation for Sinatra in Apisonator.

**How to verify:**

1. Follow the instructions on the [docs](https://opentelemetry.io/docs/languages/ruby/exporters/) to launch a Jaeger container.
2. Set `OTEL_EXPORTER_OTLP_ENDPOINT="http://localhost:4318"` and the new var  `CONFIG_OPENTELEMETRY_ENABLED=true`.
3. Work normally, e.g. send some transaction requests to the listener.
4. Visit Jaeger UI at http://localhost:16686 and verify the requests are reported.
5. Verify that not setting `CONFIG_OPENTELEMETRY_ENABLED` or setting it to `false` disables the metrics and you can't see the requests in Jaeger.

**Jira Issue:**
https://issues.redhat.com/browse/THREESCALE-9682
